### DISCUSSION TITLE
Can segments have further directory inside?

### DISCUSSION THREAD

**Discussion by dante01yoon** [feature-sliced/documentation#662] [Q&A] [answered]
Greetings.

I wonder if `shared`'s segments like lib have too many util functions inside, can we have additional folder inside to gather related functions? 

for instance, one of example in documentation [kinomore-v2](https://github.com/Shiyan7/kinomore-v2/blob/master/src/shared/api/requests/index.ts) has one more directory like `shared/api/request/index.ts`

I'm not sure FSD is strictly avoid has directory depth more than 3 (layer, slice, segments).

**Comment by illright** [comment, marked as answer]
Hi. Segments can absolutely have folders inside, FSD doesn't regulate that. You can have half of your project in one segment, it's fine. 


FSD does have recommendations for folders like `shared/lib` — resist the urge to make folders with names like `utils`, `hooks`. Those names might describe what's inside but they don't help you to navigate your code — you still have to search through every file. Alternatively, names like `dates`, `i18n` are good for folders in `shared/lib`, because if you're not looking for dates or i18n code, you can skip searching in these folders entirely.

**Reply by dante01yoon** [reply]
Thank you for quick response. I wonder FSD solves problems of finding deep nested modules by regulating each modules nested level at most 3 ( layer - slice -segment) compares to monolith file structure 🤔 .

**Reply by illright** [reply]
It tends to happen naturally because most of the time, segments end up pretty small, containing just a handful of files, and don't usually have a deep nesting structure. If you examine some [examples](https://feature-sliced.design/examples), you will see that deep nesting doesn't usually occur
